qiita_db/search.py

```diff
@@ -194,7 +200,8 @@ def __call__(self, searchstr, user):
         Metadata column names and string searches are case-sensitive
         """
         study_sql, sample_sql, meta_headers = \
-            self._parse_study_search_string(searchstr, True)
```

Comment:
This method is already incredibly complex, and running on 150 lines... I think it should likely be decomposed prior to adding in more functionality.

The method name is `_parse_study_search_string`, but isn't the removal of samples from an analysis different than parsing a study search string?